react/features/overlay/actions.js

```diff
@@ -57,3 +63,37 @@ export function setFatalError(fatalError) {
         fatalError
     };
 }
+
+/**
+ * FIXME naming is not quite accurate - came from the previous method which was
```

Comment:
I think having this session code in the overlay looks a bit odd.

Reply:
Yes. That's the part I wanted to discuss. I was thinking if it should be part of the session feature in some way.

Also I took the approach of not dispatching a failed action with `recoverable = false` as opposed to what we've discussed with @lyubomir some time ago. I would have to revert current `fatalError` logic to not be a state listener, but a middleware which captures the action.